src/dex-ui/services/DexService/dao/DAOService.ts

```diff
@@ -172,8 +175,24 @@ async function fetchGovernanceDAOs(eventTypes?: string[]): Promise<GovernanceDAO
     new ethers.utils.Interface(FTDAOFactoryJSON.abi),
     eventTypes
   );
-  return logs.map((log): GovernanceDAODetails => {
+
+  const allPromises = logs.map(async (log): Promise<GovernanceDAODetails> => {
     const argsWithName = getEventArgumentsByName<GovernanceDAOCreatedEventArgs>(log.args, ["webLinks"]);
+
+    argsWithName.daoAddress = (await DexService.fetchContractId(argsWithName.daoAddress)).toSolidityAddress();
```

Comment:
Create new variables instead of mutating the existing object (argsWithName). The async mutations can lead to issues that are difficult to trace and debug.